### PR TITLE
Truncated money request preview

### PR DESCRIPTION
**Pull request by esh-g**
### Fixed Issues

$ https://github.com/Expensify/App/issues/29459
PROPOSAL: https://github.com/Expensify/App/issues/29459#issuecomment-1759825081

### Tests

1. Click on request money.
2. Enter any amount
3. Enter a long description (more than 4 lines)
4. Make sure it is truncated in the preview

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Click on request money.
2. Enter any amount
3. Enter a long description (more than 4 lines)
4. Make sure it is truncated in the preview

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/77237602/d8886b98-672e-4dbe-8cdd-a15340c43321

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/77237602/d761d20b-061e-47d2-97d3-e540f76b57ca

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/77237602/db7c8dfe-822b-4f03-9d0c-ece7e2a00876

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/77237602/e10d0175-cf8f-4494-8ad2-bde356fbd276

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/77237602/b114757c-06e0-4e59-8a28-fdb7c8768016

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/77237602/ee0f73e0-c5f4-40bb-bb03-750892531239

</details>

### CODE REVIEW

**Comment by esh-g** [comment]
@situchan All the videos added... Hoping this can be a quick one

**Comment by esh-g** [comment]
gentle bump @situchan

**Comment by esh-g** [comment]
gentle bump @situchan

**Comment by situchan** [comment]
Not sure if this is bug or expected. text is scrollable inside preview
I think we should prevent this for platform consistency as not happening on mobile.
cc: @Expensify/design 


https://github.com/Expensify/App/assets/108292595/5fd7a4f7-1ec5-4047-b999-2aaedcd63b79

**Comment by shawnborton** [comment]
Wow, interesting. @JmillsExpensify or @trjExpensify do you have any context here? 

I think there definitely shouldn't be a scrollable area, so not sure how that happened. I would think we just truncate this one after 2-3 lines max, and ideally this component behaves the same as other preview components.

**Comment by esh-g** [comment]
@situchan I investigated this and apparently, this is an issue in `react-native-web` itself as mentioned here: https://github.com/necolas/react-native-web/issues/13#issuecomment-449417403

But I have a solution to this:
Instead of using `numberOfLines` prop to limit the text, we can set the `maxHeight` of the `View` to `numberOfLines * lineHeight`
which can look something like this:
```js
<View style={[styles.flex1, truncatedContainerStyle]}>
    {!isCurrentUserManager && props.shouldShowPendingConversionMessage && (
        <Text style={[styles.textLabel, styles.colorMuted]}>{props.translate('iou.pendingConversionMessage')}</Text>
    )}
    {shouldShowDescription && <Text style={[styles.colorMuted]}>{description}</Text>}
</View>
```
where `truncatedContainerStyle` looks like this:
```js
const truncatedContainerStyle = StyleUtils.combineStyles([
    StyleUtils.getMaximumHeight(3 * variables.fontSizeNormalHeight),
    styles.overflowHidden,
]);
```

same can be done for the merchant.

**Comment by dannymcclain** [comment]
> I think there definitely shouldn't be a scrollable area, so not sure how that happened. I would think we just truncate this one after 2-3 lines max, and ideally this component behaves the same as other preview components.

Agree with Shawn here. Truncate after 2-3 lines (I'd kinda like to see both to see how they feel) with an "..." at the end to let users know there's more once they tap into it.

**Comment by situchan** [comment]
>  we can set the `maxHeight` of the `View` to `numberOfLines` * `lineHeight`

Did you test that approach on all platforms?

**Comment by esh-g** [comment]
@situchan Yes, I did test on `iOS`, `mWeb` and `desktop/chrome`....

**Comment by situchan** [comment]
@esh-g what you linked is very old article. can you think of alternative solution, not setting maxHeight?

**Comment by esh-g** [comment]
@situchan I don't think there is an alternative...
- We cannot use the `onLayout` prop on web to get the number of lines.
- The current `numberOfLines` prop on web, translates to `-webkit-line-clamp` which cannot prevent scroll (even if overflow hidden is added).

The other option we have is to limit the number of characters in the IOU preview., which can be done easily...

**Comment by situchan** [comment]
@esh-g ok, please push changes and update screenshots accordingly with scrolling tested. Make sure to test various text length which fill 1-2 lines, 3 lines, 3+ lines

**Comment by situchan** [comment]
> Okay @situchan Just a note that there are no ellipsis with this approach, I hope that's fine?

No, ellipsis should show

**Comment by esh-g** [comment]
@situchan Then I think it's better we go for a preview with a fixed number of characters instead... because that will be predictable to add the ellipsis and I don't see any downside with that approach either...

**Comment by esh-g** [comment]
Gentle bump @situchan

**Comment by situchan** [comment]
> @situchan Then I think it's better we go for a preview with a fixed number of characters instead... because that will be predictable to add the ellipsis and I don't see any downside with that approach either...

Nope, we should not truncate characters as width / font width will vary depending on devices.

**Comment by esh-g** [comment]
@situchan I wish there was a solution to that but seeing that even react-native-web was not able to come up with a solution to the scroll problem, I think the variation in the number of lines would be at most 1-2 lines...

But still if we want the current approach with the ellipsis, then we would have to add the ellipsis to the next line like in the following image:
Screenshot 2023-10-30 at 10 16 18 PM

This is achieved by adding a state var called `subtitleHeight` that is set in the `onLayout` callback in description or merchant `Text`, then checking if `subtitleHeight > 3 * variables.fontSizeNormalHeight` and displaying ellipsis if it is...

cc @Expensify/design (to decide with the ellipsis on the next line...)

**Comment by trjExpensify** [comment]
I'm not a huge fan of adding a whole line to the preview just for the ellipsis, what do you think @dannymcclain?

**Comment by dannymcclain** [comment]
@trjExpensify yeah I'm not a fan of that either. If we can't reliably truncate based on lines I think I'd rather do it based on character count like was suggested [here](https://github.com/Expensify/App/pull/30240#issuecomment-1781346287) (I think?). I realize that might give us less control over the number of lines that show up but we could probably choose a character count that we're pretty confident wouldn't span to 3 lines. And it would likely look a LOT cleaner. What do you think?

**Comment by trjExpensify** [comment]
Yeah, are you suggesting we have an ellipsis still to "signal" the truncation (which I think is important) but after X characters such that it's contained to two lines. I agree that's better than: 

1. no ellipsis to signal the text is truncated
2. an ellipsis on its own line

**Comment by esh-g** [comment]
So do we agree to go ahead with a character limit? If yes, then what is the amount of characters we are thinking?

**Comment by trjExpensify** [comment]
If we've exhausted our options for the "preferred" method of using numberOfLines then, yeah. 

I honestly don't think descriptions for expenses are super long on average. They're normally pretty snappy, so something like 40 seems adequate to keep to two lines, but I think we might need to see it. :)

**Comment by esh-g** [comment]
@situchan Should we go for the approach of limiting the characters then?

**Comment by esh-g** [comment]
gentle bump @situchan

**Comment by esh-g** [comment]
bump @situchan

**Comment by situchan** [comment]
> @situchan Should we go for the approach of limiting the characters then?

Personally I don't like that approach.

@joelbettner what do you suggest? As you're assigned on the linked issue

**Comment by esh-g** [comment]
gentle bump @joelbettner

**Comment by joelbettner** [comment]
I think we should limit based on character count.

**Comment by esh-g** [comment]
@situchan Should we look at a limit of 40-50 words? What should be the exact number?

**Comment by esh-g** [comment]
@situchan should we go ahead with 40 char limit as discussed here: https://expensify.enterprise.slack.com/archives/C01GTK53T8Q/p1701891930152869?thread_ts=1701890357.322829&channel=C01GTK53T8Q&message_ts=1701891930.152869

**Comment by situchan** [comment]
> @situchan should we go ahead with 40 char limit as discussed here: https://expensify.enterprise.slack.com/archives/C01GTK53T8Q/p1701891930152869?thread_ts=1701890357.322829&channel=C01GTK53T8Q&message_ts=1701891930.152869

I think so as no objections so far

**Comment by esh-g** [comment]
@situchan I think it would be better if we go with something like 120 characters because 40 doesn't even seem to cover one line...
Screenshot 2023-12-16 at 12 25 44 AM
Screenshot 2023-12-16 at 12 25 27 AM

**Comment by esh-g** [comment]
bump @situchan

**Comment by dannymcclain** [comment]
@esh-g 120 characters looks good to me. I could also see us going down to 80–100 characters just to make super sure we don't go over 3 lines. @trjExpensify any strong feelings here? 40 does seem pretty short. cc @Expensify/design

**Comment by trjExpensify** [comment]
Yeah, I'm not passionate. I'll defer to you man!

**Comment by shawnborton** [comment]
I like two lines personally, that seems reasonable?

**Comment by dannymcclain** [comment]
@esh-g Can we try 80 characters and see how that feels? I think that will bring us to about 2 lines.

**Comment by esh-g** [comment]
Yes @dannymcclain that sounds good as well! Making the change now

**Comment by shawnborton** [comment]
I think we can also use flex styling to just indicate that we only want 2 lines as well?

**Comment by esh-g** [comment]
@shawnborton Do you mean cutting out the description so that only first 2 lines are visible? We already discussed that and the problem was that there would be no ellipsis in the text..

**Comment by shawnborton** [comment]
I thought we could use the baked in `ellipsizeMode`? https://reactnative.dev/docs/text#ellipsizemode

**Comment by esh-g** [comment]
@shawnborton Yes we can use that on native but it is broken for react-native-web. If you have more than one line, it allows the user to scroll on web. This is the reason we decided to cap characters instead of lines..

**Comment by shawnborton** [comment]
Ah got it, thanks for clarifying!

**Comment by esh-g** [comment]
btw, I updated it to 83 because the lodash truncate function also counts the 3 ellipsis as characters so, if we had set it to 80, we would only get 77 characters.
Screenshot 2023-12-19 at 11 00 02 PM

**Comment by esh-g** [comment]
Does the change look good? @situchan

**Comment by situchan** [comment]
Looks good. Please also test with various width of characters including emojis.

**Comment by esh-g** [comment]
@situchan Seems like we can fit a bit more emojis than text... but is this something we should change? Because I wouldn't say it is realistic to use only emojis and there is no way to strictly enforce the 2 line limit for different characters...

Screenshot 2023-12-21 at 11 18 57 PM

**Comment by esh-g** [comment]
bump @situchan

**Comment by situchan** [comment]
I will complete review by tomorrow

File: src/components/ReportActionItem/MoneyRequestPreview.js [hunk 1]
```diff
@@ -151,8 +152,9 @@ function MoneyRequestPreview(props) {
     // Pay button should only be visible to the manager of the report.
     const isCurrentUserManager = managerID === sessionAccountID;
 
-    const {amount: requestAmount, currency: requestCurrency, comment: requestComment, merchant: requestMerchant} = ReportUtils.getTransactionDetails(props.transaction);
-    const description = requestComment;
+    const {amount: requestAmount, currency: requestCurrency, comment: requestComment, merchant} = ReportUtils.getTransactionDetails(props.transaction);
+    const description = truncate(requestComment, {length: 83});
```
Comment (situchan) [inline src/components/ReportActionItem/MoneyRequestPreview.js:156]: define this number in CONST

Comment (esh-g) [reply src/components/ReportActionItem/MoneyRequestPreview.js]: @situchan Done!

**Comment by situchan** [comment]
83 is the best value?

3 lines on iOS:

ios

**Comment by situchan** [comment]
If that's fine, I will go ahead given that there's no perfect solution to be responsive to all platforms.

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

android


</details>

<details>
<summary>Android: mWeb Chrome</summary>

mchrome


</details>

<details>
<summary>iOS: Native</summary>

ios


</details>

<details>
<summary>iOS: mWeb Safari</summary>

msafari


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web


</details>

<details>
<summary>MacOS: Desktop</summary>

desktop


</details>

**Review by situchan — APPROVED** [review]

**Comment by situchan** [comment]
@esh-g I don't see scroll issue in other places of the app where `numberOfLines` is set.
i.e. share code page

https://github.com/Expensify/App/blob/d5f5856b610cc39d5b5f2b054dab67bb74d66922/src/components/QRShare/index.js#L58

Screenshot 2024-01-04 at 7 21 02 PM

**Review by joelbettner — APPROVED** [review]

File: src/components/ReportActionItem/MoneyRequestPreview.js [hunk 2]
```diff
@@ -151,8 +152,9 @@ function MoneyRequestPreview(props) {
     // Pay button should only be visible to the manager of the report.
     const isCurrentUserManager = managerID === sessionAccountID;
 
-    const {amount: requestAmount, currency: requestCurrency, comment: requestComment, merchant: requestMerchant} = ReportUtils.getTransactionDetails(props.transaction);
-    const description = requestComment;
+    const {amount: requestAmount, currency: requestCurrency, comment: requestComment, merchant} = ReportUtils.getTransactionDetails(props.transaction);
```
Comment (joelbettner) [inline src/components/ReportActionItem/MoneyRequestPreview.js:155]: Why are we changing `, merchant: requestMerchant}` to just `, merchant}`??

Comment (situchan) [reply src/components/ReportActionItem/MoneyRequestPreview.js:155]: `requestMerchant` is defined below Line 157

Comment (joelbettner) [reply src/components/ReportActionItem/MoneyRequestPreview.js:155]: Then do we even need the `,merchant}` in here anymore? The value for `merchant` that gets returned from `getTransactionDetails` isn't being used anywhere anymore...is it?

**Review by joelbettner — APPROVED** [review]

**Comment by esh-g** [comment]
> @esh-g I don't see scroll issue in other places of the app where `numberOfLines` is set. i.e. share code page
> 
> https://github.com/Expensify/App/blob/d5f5856b610cc39d5b5f2b054dab67bb74d66922/src/components/QRShare/index.js#L58
> 

It definitely is very strange behaviour for `react-native-web` maybe because the component was already scrollable in case of the money request action, it could not block any scroll events so we were able to scroll in the money request description but not here...

**Pull request merged**